hr_timesheet_timetracker/models/account_analytic_line.py

```diff
@@ -0,0 +1,363 @@
+    def _start_timetracking(self, timestamp):
+        self.write({
+            'timetracker_started_at': timestamp,
+        })
+
+    @api.multi
+    def _stop_timetracking(self, timestamp):
+        self.write({
+            'timetracker_stopped_at': timestamp,
+        })
+
+    @api.multi
+    def toggle_timetracker(self):
+        now = fields.Datetime.now()
+
+        affected_lines = []
+        for line in self:
+            if not line.can_use_timetracker and \
+                    not line.check_access_rule('write'):
+                raise UserError(_(
+                    'You\'re not allowed to alter timetracking of another'
+                    ' employee'
+                ))
+
+            if line.is_timetracker_running:
+                # In case tracking is active, just stop it
+                line._stop_timetracking(now)
+                affected_lines.append(line)
+            elif not line.timetracker_started_at and \
+                    line.unit_amount == 0.0:
+                # In case tracking was not active and duration is 0
+                # (i.e. new entry), start tracking, optionally stopping
+                # current one
+                currently_timetracked = line.get_currently_timetracked()
+                currently_timetracked._stop_timetracking(now)
+                affected_lines.extend(currently_timetracked)
+                line._start_timetracking(now)
+                affected_lines.append(line)
+            else:
+                # Otherwise make a copy and start tracking it,
+                # optionally stopping current one
+                currently_timetracked = line.get_currently_timetracked()
+                currently_timetracked._stop_timetracking(now)
+                affected_lines.extend(currently_timetracked)
+                new_timetracked_line = line.copy({
+                    'timetracker_started_at': None,
+                    'timetracker_stopped_at': None,
+                    'unit_amount': 0,
+                })
+                new_timetracked_line._start_timetracking(now)
+                affected_lines.append(new_timetracked_line)
+
+        return affected_lines
+
+    @api.multi
+    def action_timetracker(self):
+        self.toggle_timetracker()
+
+        return {
+            'type': 'ir.actions.act_view_reload',
+        }
+
+    @api.multi
+    def action_reset_timetracked_duration(self):
+        self.reset_timetracked_duration()
+
+        return {
+            'type': 'ir.actions.act_view_reload',
+        }
+
+    @api.model
+    @api.returns('self')
+    def get_currently_timetracked(self, users=None):
+        users = users or self.user_id or self.env.user
+        return self.search([
+            ('user_id', 'in', users.ids),
+            ('timetracker_started_at', '!=', False),
+            ('timetracker_stopped_at', '=', False),
+        ])
+
+    @api.multi
+    def reset_timetracked_duration(self):
+        for line in self:
+            line.write({
```

Comment:
add also `'unit_amount': line._calculate_duration(),`

Reply:
@nikul-serpentcs This won't be correct, see below